x-pack/plugins/security_solution/cypress/integration/data_sources/sourcerer.spec.ts

```diff
@@ -51,88 +56,175 @@ describe('Sourcerer', () => {
-  // has always been skipped with intentions to fix
-  describe.skip('Timeline scope', () => {
+
+  describe('Timeline scope', () => {
     beforeEach(() => {
       cy.clearLocalStorage();
-      loginAndWaitForPage(HOSTS_URL);
+      loginAndWaitForPage(TIMELINES_URL);
     });
 
-    const alertPatterns = ['.siem-signals-default'];
-    const rawPatterns = ['auditbeat-*'];
-    const allPatterns = [...alertPatterns, ...rawPatterns];
-
-    it('Radio buttons select correct sourcerer patterns', () => {
+    it('correctly loads SIEM data view before and after signals index exists', () => {
       openTimelineUsingToggle();
       openSourcerer('timeline');
-      allPatterns.forEach((ss) => isSourcererSelection(ss, 'timeline'));
-      clickTimelineRadio('raw');
-      rawPatterns.forEach((ss) => isSourcererSelection(ss, 'timeline'));
-      alertPatterns.forEach((ss) => isNotSourcererSelection(ss, 'timeline'));
-      clickTimelineRadio('alert');
-      alertPatterns.forEach((ss) => isSourcererSelection(ss, 'timeline'));
-      rawPatterns.forEach((ss) => isNotSourcererSelection(ss, 'timeline'));
+      isDataViewSelection(siemDataViewTitle);
+      openAdvancedSettings();
+      isSourcererSelection(`auditbeat-*`);
+      isNotSourcererSelection(`${DEFAULT_ALERTS_INDEX}-default`);
+      isSourcererOptions(
+        [...DEFAULT_INDEX_PATTERN, `${DEFAULT_ALERTS_INDEX}-default`].filter(
+          (pattern) => pattern !== 'auditbeat-*'
+        )
+      );
+      waitForAlertsIndexToExist();
+      isSourcererOptions(DEFAULT_INDEX_PATTERN.filter((pattern) => pattern !== 'auditbeat-*'));
+      isNotSourcererOption(`${DEFAULT_ALERTS_INDEX}-default`);
     });
 
-    it('Adding an option results in the custom radio becoming active', () => {
-      openTimelineUsingToggle();
-      openSourcerer('timeline');
-      isNotCustomRadio();
-      clickOutOfSourcererTimeline();
-      const luckyOption = 'logs-*';
-      setSourcererOption(luckyOption, 'timeline');
-      openSourcerer('timeline');
-      isCustomRadio();
+    describe('Modified badge', () => {
+      it('Selecting new data view does not add a modified badge', () => {
+        openTimelineUsingToggle();
+        cy.get(SOURCERER.badgeModified).should(`not.exist`);
+        openSourcerer('timeline');
+        cy.get(SOURCERER.badgeModifiedOption).should(`not.exist`);
+        openDataViewSelection();
+        isKibanaDataViewOption(dataViews);
+        cy.get(SOURCERER.selectListDefaultOption).should(`contain`, siemDataViewTitle);
+        cy.get(SOURCERER.selectListOption).contains(dataViews[1]).click();
+        isDataViewSelection(dataViews[1]);
+        saveSourcerer();
+        cy.get(SOURCERER.badgeModified).should(`not.exist`);
+        openSourcerer('timeline');
+        cy.get(SOURCERER.badgeModifiedOption).should(`not.exist`);
+      });
+
+      it('shows modified badge when index patterns change and removes when reset', () => {
```

Comment:
I would also split this into 2 different tests.

Reply:
Like this?? i still have to do the modification so I am repeating behavior in the following test. is that ok?
```
it('shows modified badge when index patterns change', () => {
  openSourcerer();
  openDataViewSelection();
  cy.get(SOURCERER.selectListOption).contains(dataViews[1]).click();
  isDataViewSelection(dataViews[1]);
  openAdvancedSettings();
  const patterns = dataViews[1].split(',');
  deselectSourcererOptions([patterns[0]]);
  saveSourcerer();
  cy.get(SOURCERER.badgeModified).should(`exist`);
  openSourcerer();
  cy.get(SOURCERER.badgeModifiedOption).should(`exist`);
});

it('removes modified badge when index patterns are reset', () => {
  openSourcerer();
  openDataViewSelection();
  cy.get(SOURCERER.selectListOption).contains(dataViews[1]).click();
  isDataViewSelection(dataViews[1]);
  openAdvancedSettings();
  const patterns = dataViews[1].split(',');
  deselectSourcererOptions([patterns[0]]);
  saveSourcerer();
  openSourcerer();
  resetSourcerer();
  saveSourcerer();
  cy.get(SOURCERER.badgeModified).should(`not.exist`);
  openSourcerer();
  cy.get(SOURCERER.badgeModifiedOption).should(`not.exist`);
  isDataViewSelection(siemDataViewTitle);
});
```